ballista/scheduler/src/state/task_manager.rs

```diff
@@ -505,9 +505,8 @@ impl<T: 'static + AsLogicalPlan, U: 'static + AsExecutionPlan> TaskManager<T, U>
 
             Ok(Some(Arc::new(guard.deref().clone())))
         } else {
-            let graph = self.state.get_execution_graph(job_id).await?;
-
-            Ok(graph.map(Arc::new))
+            // We drop the `ExecutionGraph` after job is completed so always return `None` here
+            Ok(None)
```

Comment:
Is this gonna interfere with `get_query_status` API?
https://github.com/search?q=repo%3Acoralogix%2Feng-svc-dataprime-api%20getQueryStatus&type=code

Reply:
No, we store the status separately in the state. This just drops the execution graph